app/views/teachers/RequestQuoteView.coffee

```diff
@@ -0,0 +1,258 @@
+    properties = @trialRequest.get('properties')
+    if properties
+      forms.objectToForm(@$('#request-form'), properties)
+      commonLevels = _.map @$('[name="educationLevel"]'), (el) -> $(el).val()
+      submittedLevels = properties.educationLevel or []
+      otherLevel = _.first(_.difference(submittedLevels, commonLevels)) or ''
+      @$('#other-education-level-checkbox').attr('checked', !!otherLevel)
+      @$('#other-education-level-input').val(otherLevel)
+      
+    # apply changes from local storage
+    obj = storage.load(FORM_KEY)
+    if obj
+      @$('#other-education-level-checkbox').attr('checked', obj.otherChecked)
+      @$('#other-education-level-input').val(obj.otherInput)
+      forms.objectToForm(@$('#request-form'), obj, { overwriteExisting: true })
+
+  onChangeRequestForm: ->
+    # save changes to local storage
+    obj = forms.formToObject(@$('form'))
+    obj.otherChecked = @$('#other-education-level-checkbox').is(':checked')
+    obj.otherInput = @$('#other-education-level-input').val()
+    storage.save(FORM_KEY, obj, 10)
+
+  onSubmitRequestForm: (e) ->
+    e.preventDefault()
+    form = @$('#request-form')
+    attrs = forms.formToObject(form)
+    
+    # custom other input logic (also used in form local storage save/restore)
+    if @$('#other-education-level-checkbox').is(':checked')
+      attrs.educationLevel.push(@$('#other-education-level-input').val())
+      
+    forms.clearFormAlerts(form)
+    requestFormSchema = if me.isAnonymous() then requestFormSchemaAnonymous else requestFormSchemaLoggedIn
+    result = tv4.validateMultiple(attrs, requestFormSchemaAnonymous)
+    error = false
+    if not result.valid
+      forms.applyErrorsToForm(form, result.errors)
+      error = true
+    if not forms.validateEmail(attrs.email)
+      forms.setErrorToProperty(form, 'email', 'Invalid email.')
+      error = true
+    if not _.size(attrs.educationLevel)
+      forms.setErrorToProperty(form, 'educationLevel', 'Check at least one.')
+      error = true
+    if error
+      forms.scrollToFirstError()
+      return
+    @trialRequest = new TrialRequest({
+      type: 'course'
+      properties: attrs
+    })
+    if me.get('role') is 'student' and not me.isAnonymous()
+      modal = new ConfirmModal({
+        title: ''
+        body: "<p>#{$.i18n.t('teachers_quote.conversion_warning')}</p><p>#{$.i18n.t('teachers_quote.learn_more_modal')}</p>"
+        confirm: $.i18n.t('common.continue')
+        decline: $.i18n.t('common.cancel')
+      })
+      @openModalView(modal)
+      modal.once 'confirm', @saveTrialRequest, @
+    else
+      @saveTrialRequest()
+    
+  saveTrialRequest: ->
+    @trialRequest.notyErrors = false
+    @$('#submit-request-btn').text('Sending').attr('disabled', true)
+    @trialRequest.save()
+    @trialRequest.on 'sync', @onTrialRequestSubmit, @
+    @trialRequest.on 'error', @onTrialRequestError, @
+    me.setRole @trialRequest.get('properties').role.toLowerCase(), true
```

Comment:
We shouldn't update the user role until the trial request has been saved successfully.